File: app/src/main/java/ru/dmitrybochkov/rates/di/AppComponent.java
package ru.dmitrybochkov.rates.di;

import javax.inject.Singleton;

import dagger.Component;
import ru.dmitrybochkov.rates.view_models.RatesViewModel;

/**
 * Created by Dmitry Bochkov on 14.11.2019.
 */
@Component(modules = {AppModule.class, DataModule.class, NetworkServiceModule.class, UseCasesModule.class})
@Singleton
public interface AppComponent {


    void inject(RatesViewModel ratesViewModel);
}

File: app/src/main/java/ru/dmitrybochkov/rates/di/AppComponent.kt
package ru.dmitrybochkov.rates.di

import dagger.Component
import ru.dmitrybochkov.rates.view_models.RatesViewModel
import javax.inject.Singleton

/**
 * Created by Dmitry Bochkov on 14.11.2019.
 */
@Component(modules = [AppModule::class, DataModule::class, NetworkServiceModule::class, UseCasesModule::class])
@Singleton
interface AppComponent {


    fun inject(ratesViewModel: RatesViewModel)
}

